### ISSUE TITLE
ENH: stats: need geometric zscore

### ISSUE THREAD

**Issue by qbarthelemy** [scipy/scipy#14816]
### Describe the solution you'd like.

Scipy already supports `gmean` and `gstd`, the geometric versions of `mean` and `std`.
This feature request concerns `gzscore`, the geometric version of `zscore`, useful for log-normally distributed samples.

### Describe alternatives you've considered.

The definition of geometric zscore is [[ref]](https://en.wikipedia.org/wiki/Geometric_standard_deviation#Geometric_standard_score):
`gz = log(x/gm) / log(gs)`,
where `gm` is the geometric mean, and `gs` the geometric standard deviation.
But this naive implementation is slow (see below).

The implementation added in PR #14817 re-uses `zscore`, because `gzscore` can be simpy re-written as the zscore of log-values.
This implementation is fast (see below), and it inherits from all options available in `zscore`: `axis`, `ddof` and `nan_policy`.

### Additional context (e.g. screenshots, GIFs)

```
import time
import numpy as np
from numpy import ma
from scipy.stats import gmean, gstd, zscore

def gzscore_naive(a, axis=0, ddof=0):  # OK only for axis=0
    gm = gmean(a, axis=axis)
    gs = gstd(a, axis=axis, ddof=ddof)
    return np.log(a / gm) / np.log(gs)

def gzscore(a, axis=0, ddof=0, nan_policy='propagate'):
    a = np.asanyarray(a)
    log = ma.log if isinstance(a, ma.MaskedArray) else np.log
    return zscore(log(a), axis=axis, ddof=ddof, nan_policy=nan_policy)

def compute_time(func, axis=0, ddof=0, n_reps=10, n_dim0=1000, n_dim1=5000):
    rs = np.random.RandomState(42)
    time_elapsed = np.zeros((n_reps))
    for r in range(n_reps):
        x = np.abs(rs.randn(n_dim0, n_dim1))
        time_start = time.time()
        func(x, axis=axis, ddof=ddof)
        time_elapsed[r] = (time.time() - time_start)
    print('Computation time = {0:.3f} +/- {1:.3f}'.format(time_elapsed.mean(),
                                                          time_elapsed.std()))

axis, ddof = 0, 0
compute_time(gzscore_naive, axis, ddof)
compute_time(gzscore, axis, ddof)
```
gives
```
Computation time = 0.180 +/- 0.002
Computation time = 0.115 +/- 0.002
```

**Comment by tupui** [comment]
Hi, thanks for the proposal. Sounds like a sensible and on the scope enhancement.

**Comment by qbarthelemy** [comment]
Solved with #14817.
